migration-engine/connectors/mongodb-migration-connector/src/client_wrapper.rs

```diff
@@ -67,8 +69,16 @@ impl Client {
                     continue; // do not introspect or diff these
                 }
 
-                let path = index.keys;
-                schema.push_index(collection_id, name, is_unique, path);
+                let path = if preview_features.contains(PreviewFeature::ExtendedIndexes) {
+                    index.keys
+                } else {
+                    index.keys.iter().fold(Document::new(), |mut acc, (k, _)| {
+                        acc.insert(k, Bson::Int32(1));
+                        acc
+                    })
+                };
```

Comment:
I think we're missing a test about changing the direction of an index (asc -> desc or the other way). I don't think we're diffing that in this PR.